From 83633c9f36c69a7a3d3835350b7acfc92c7ee870 Mon Sep 17 00:00:00 2001
From: Sean Paul <seanpaul@chromium.org>
Date: Tue, 9 Jan 2018 13:53:51 -0500
Subject: [PATCH 0992/2313] UPSTREAM: drm/i915: Don't allow HDCP on PORT E/F

Port E doesn't have HDCP support, and Port F is disabled. Don't setup
the hdcp shim on those.

Reviewed-by: Daniel Vetter <daniel@ffwll.ch>
Signed-off-by: Sean Paul <seanpaul@chromium.org>
Link: https://patchwork.freedesktop.org/patch/msgid/20180109185401.16911-1-seanpaul@chromium.org
(cherry picked from commit e6446bc454c608375c1195edfd7e2ef5529c14ef)

Change-Id: I2988ba44800ffd77cbff7fca0d8c19f57e339d47
---
 drivers/gpu/drm/i915/intel_hdmi.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_hdmi.c b/drivers/gpu/drm/i915/intel_hdmi.c
index f9fdf69..e60d1f9 100644
--- a/drivers/gpu/drm/i915/intel_hdmi.c
+++ b/drivers/gpu/drm/i915/intel_hdmi.c
@@ -2282,7 +2282,8 @@ void intel_hdmi_init_connector(struct intel_digital_port *intel_dig_port,
 
 	intel_hdmi_add_properties(intel_hdmi, connector);
 
-	if (INTEL_GEN(dev_priv) >= 9) {
+	/* PORT E doesn't have HDCP, and PORT F is disabled */
+	if (INTEL_GEN(dev_priv) >= 9 && port < PORT_E) {
 		int ret = intel_hdcp_init(intel_connector,
 					  &intel_hdmi_hdcp_shim);
 		if (ret)
-- 
2.7.4

